dataduct/s3/utils.py

```diff
@@ -119,6 +129,46 @@ def upload_dir_to_s3(s3_path, local_path, filter_function=None):
         local_path(file_path): Input path of the file to be uploaded
         filter_function(function): Function to filter out directories
     """
+
+    def multipart_upload(key_string, local_file_path):
+        username = os.path.basename(os.path.expanduser('~'))
+        directory = '/tmp/multipart_upload_{}'.format(username)
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+
+        uuid_string = str(uuid.uuid1())
+        prefix = os.path.join(directory, 'tmp_upload_{}'.format(uuid_string))
+
+        # split file into parts
+        split = ["split",
+                 "-b%s" % CHUNK_SIZE,
+                 local_file_path,
+                 prefix]
+        subprocess.check_call(split)
+        files = glob.glob('{}*'.format(prefix))
+
+        try:
+            mpu = bucket.initiate_multipart_upload(key_string)
+            logger.info('Multipart uploading into {} ...'.format(key_string))
+            start_time = time.time()
+            for i, file_part in enumerate(files):
```

Comment:
May be add something like a progress bar here to show status of upload 

https://github.com/rasbt/pyprind